ddev/tests/repo/test_core.py

```diff
@@ -42,115 +42,417 @@ def test_valid(self, local_repo):
         assert isinstance(integration, Integration)
 
 
-class TestIteration:
-    def test_only_integrations(self, local_repo):
-        repo = Repository(local_repo.name, str(local_repo))
+class TestIterationOnlyIntegrations:
+    def test_default_changed(self, repository):
+        repo = Repository(repository.path.name, str(repository.path))
 
         integration_names = []
-        for path in local_repo.iterdir():
+        for path in repository.path.iterdir():
             if (path / 'manifest.json').is_file():
                 integration_names.append(path.name)
         integration_names.sort()
 
+        integration = integration_names[0]
+        (repository.path / integration / 'foo.txt').touch()
+
+        changed = [integration]
         integrations = list(repo.integrations.iter())
-        assert [integration.name for integration in integrations] == integration_names
+
+        assert [integration.name for integration in integrations] == changed
         assert list(repo.integrations.iter()) == integrations
 
+    def test_selection(self, local_repo):
+        repo = Repository(local_repo.name, str(local_repo))
+
+        integration_names = []
+        for path in local_repo.iterdir():
+            if (path / 'manifest.json').is_file():
+                integration_names.append(path.name)
+        integration_names.sort()
```

Comment:
These apply somewhat similarly to all test classes.
1. This setup is repeated in all tests in the class. Might be worth factoring it out.
2. Might as well use `Integration.is_integration`, since this is equivalent?

Also, in general, this looks like a testing matrix manually rolled out for two parameters:
1. The filtering criteria for the integrations.
2. The selection (default changed, selection and all).

I.e. something worth parameterizing. As they are now, modifying something in these tests looks painful and error prone.

Reply:
How would you recommend parameterizing these?

Reply:
Some ideas: https://github.com/DataDog/integrations-core/compare/ofek/i...alopez/parametrization-suggestion

Reply:
https://github.com/DataDog/integrations-core/pull/13622